micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationContextAssert.java

```diff
@@ -197,6 +200,24 @@ else if (!extraKeys.isEmpty()) {
         return (SELF) this;
     }
 
+    /**
+     * Verifies that the Observation key-value keys are a subset of the given set of keys.
+     */
+    public SELF hasSubsetOfKeys(String... keys) {
+        isNotNull();
+        Set<String> actualKeys = new LinkedHashSet<>(allKeys());
+        Set<String> expectedKeys = new LinkedHashSet<>(Arrays.asList(keys));
+
+        List<String> extra = stream(actualKeys).filter(actualElement -> !expectedKeys.contains(actualElement))
```

Comment:
I'm thinking if we can reuse a built-in assertion from assertJ, it has contains... methods.

Reply:
Probably. I think I copied this from the implementation one of assertJ's methods. Let's take a look after RC1.